### DISCUSSION TITLE
Terminal.app `Command+L` equivalent in Kitty? (clear last command output)

### DISCUSSION THREAD

**Discussion by tmillr** [kovidgoyal/kitty#6040] [Q&A]
I'd use this shortcut all the time in Apple's Terminal.app to keep the screen + scrollback both tidy and from getting out of hand, but there doesn't seem to be an equivalent default shortcut nor action in kitty? How can I achieve this? Thanks

**Comment by kovidgoyal** [comment]
https://sw.kovidgoyal.net/kitty/conf/#shortcut-kitty.Reset-the-terminal

**Reply by tmillr** [reply]
> https://sw.kovidgoyal.net/kitty/conf/#shortcut-kitty.Reset-the-terminal

_Edit: below, by "scrollback" I mean everything, or the screen + scrollback, all text in the buffer, etc. Apologies if that's not what scrollback actually means or refers to._

Thank you. I already tried using this, but I couldn't get it to work how I wanted it to.

Here's what I last tried:

```
map cmd+l combine : scroll_to_prompt -1 : clear_terminal to_cursor active 🍎
```

...but this just clears everything (i.e. the entire screen and all of the scrollback apparently). Also, it does so in **_every single kitty tab_** that I have open? Including those in different OS windows, I believe, which confuses me a bit (especially because of the `active` passed to `clear_terminal`).

> I don't use (multiple) kitty windows, at least not at the moment.

If I'm not mistaken, and based off of my experience using it just now, `clear_terminal to_cursor` clears **_from_** the very "top" (i.e. beginning) of the scrollback down to the cursor line? If that's the case, I don't think that will help me as I'm actually looking for something that clears from the very "bottom" instead as I only want to clear the previous command and its output (i.e. from its prompt until the last line of its output, inclusive).

> P.S. And I'm assuming that kitty simply ignores the 🍎 apple emoji at the end of the map directive above? That is how I copied it from the online docs/website and I'm assuming it is ignored and just for decoration?

**Reply by page-down** [reply]
> I only want to clear the previous command and its output ...

If you only want to keep the last prompt (waiting for input), then you can use the default shortcut `cmd+k`, but I don't think that's what you want.

If you want to keep the current prompt and all history except for the last entered command and output, there is no such action now, but it can be done with the existing infrastructure. (Probably needs to be implemented in c.)

> ... I'm assuming that kitty simply ignores the 🍎 apple emoji ...

No, don't paste that into the config file. It's been on my list for a long time and I wanted to remove that trailing apple emoji when copying (javascript).

**Comment by kovidgoyal** [comment]
You want to edit the contents of the scrollback removing the last command and its output? There is no such functionality in kitty. Never felt the need for it, since browsing scrollback is very fast efficient.

**Reply by edwardloveall** [reply]
Yes, that's true Kitty is quite efficient, but lots of output can make a _human_ slow. For example typing the wrong command could print many lines of traceback and debugging information because of a simple typo. It's nice to be able to clean that up so you can focus on the task and see the previous (good) commands and output.

**Reply by tmillr** [reply]
Yeah, that was one of my favorite features of Apple Terminal. That and the seamless automatic sessions and restoration.

I tend to use many tabs. My old workflow would be to go through all of the scrollback before closing a tab, and this shortcut would help me do that both at once and on an ongoing basis. Many one-shot commands like viewing the current directory or listing directory contents are temporary, and I don't need them to stick around polluting the screen and scroll back.

Maybe it was a bad workflow and habit to have, but I got a lot of use out of that shortcut.

Is it at least possible to hack up a simple solution via a kitten?

**Reply by kovidgoyal** [reply]
sure, use get-text and send-text. First get the last command output, then get all text. Remove the last command output, then clear the screen and send the edited text to replace it.

**Comment by kovidgoyal** [comment]
There you go: https://github.com/kovidgoyal/kitty/commit/c86ec79e32412f1c7165897490bd15ab4b4da4ce
